misk-core/src/main/kotlin/misk/sampling/RateLimiter.kt

```diff
@@ -118,6 +118,7 @@ class RateLimiter private constructor(
     unit: TimeUnit,
     timeout: Long
   ): Long {
+    if (permitsPerSecond == 0L) return 0L
```

Comment:
I think we could also short-circuit on negatives to make it a bit easier to understand on those edge cases. i.e.

```kotlin
if (permitsPerSecond <= 0L) return 0L
```

Tests for that scenario would be good as well.